core/libp2p/transport/tcp/tcp_connection.hpp

```diff
@@ -8,64 +8,171 @@
 
 #define BOOST_ASIO_NO_DEPRECATED
 
-#include "libp2p/transport/connection.hpp"
+#include "libp2p/connection/raw_connection.hpp"
 
 #include <array>
 #include <boost/asio.hpp>
+#include <boost/noncopyable.hpp>
+#include <ufiber/ufiber.hpp>
+#include "libp2p/multi/multiaddress.hpp"
+#include "libp2p/transport/tcp/tcp_util.hpp"
 
 namespace libp2p::transport {
 
   /**
    * @brief boost::asio implementation of TCP connection (socket).
+   * @tparam Executor boost asio executor type. Can be a strand, io context or
+   * custom.
    */
-  class TcpConnection : public Connection,
-                        public std::enable_shared_from_this<TcpConnection> {
+  template <typename Executor = boost::asio::io_context::executor_type>
+  class TcpConnection
+      : public connection::RawConnection,
+        public std::enable_shared_from_this<TcpConnection<Executor>>,
+        private boost::noncopyable {
    public:
-    using Buffer = kagome::common::Buffer;
     using Tcp = boost::asio::ip::tcp;
     using ResolverResultsType = Tcp::resolver::results_type;
+    using executor_t = Executor;
+    using yield_t = ufiber::yield_token<Executor>;
```

Comment:
Are you sure those types should be public? Maybe move some of them to private?

Reply:
yeah, nothing wrong with that